raiden/raiden_service.py

```diff
@@ -635,6 +651,214 @@ def settle(self, token_address, partner_address):
+
+        If the connection manager has no funds, this is a noop.
+        """
+        # not initialized
+        if self.funds <= 0:
+            return
+        with self.lock:
+            remaining = self.funds_remaining
+            initial = self.initial_funding_per_partner
+            joining_funds = min(
+                partner_deposit,
+                remaining,
+                initial
+            )
+            if joining_funds <= 0:
+                return
+
+            self.raiden.api.deposit(
+                self.token_address.encode('hex'),
+                partner_address.encode('hex'),
+                joining_funds
+            )
+            log.debug(
+                "joined a channel!",
+                funds=joining_funds,
+                me=pex(self.raiden.address),
+                partner=pex(partner_address)
+            )
+
+    def _retry_connect(self):
+        """Will be called when new channels in the token network are detected.
+        If the minimum number of channels was not yet established, it will try
+        to open new channels.
+
+        If the connection manager has no funds, this is a noop.
+        """
+        # not initialized
+        if self.funds <= 0:
+            return
+        with self.lock:
+            if self.funds_remaining <= 0:
+                return
+            if len(self.open_channels) >= self.initial_channel_target:
+                return
+            for partner in self.find_new_partners(self.initial_channel_target - len(self.open_channels)):
+                self.raiden.api.open(
+                    self.token_address.encode('hex'),
+                    partner.encode('hex')
+                )
+                self.raiden.api.deposit(
+                    self.token_address.encode('hex'),
+                    partner.encode('hex'),
+                    self.initial_funding_per_partner
+                )
+
+    def find_new_partners(self, number):
+        """Search the token network for potential channel partners.
+        Args:
+            number (int): number of partners to return
+        """
+        known = set([c.partner_address for c in self.open_channels])
+        known = known.union({self.__class__.BOOTSTRAP_ADDR_BIN})
+        known = known.union({self.raiden.address})
+        available = set(self.manager.channelgraph.graph.nodes()) - known
+
+        available = self._select_best_partners(available)
+        return available[:number]
+
+    def _select_best_partners(self, partners):
+        # FIXME: use a proper selection strategy
+        return list(partners)
+
+    @property
+    def initial_funding_per_partner(self):
+        """The calculated funding per partner depending on configuration and
+        overall funding of the ConnectionManager.
+        """
+        return int(
+                self.funds * (1 - self.joinable_funds_target)
+                / self.initial_channel_target
+                )
```

Comment:
please close `)` right under `return` just like you do in the `wants_more_channels()` function below